src/Materials/babylon.effect.ts

```diff
@@ -595,6 +634,18 @@
             return changed;
         }
 
+        public bindUniformBuffer(buffer: WebGLBuffer, name: string): void {
+            if (Effect._baseCache[this._uniformBuffersNames[name]] === buffer) {

```

Comment:
Here we cannot avoid it, since uniform buffers have binding points (see https://paroj.github.io/gltut/Positioning/UBOContextBinding.svg). Those binding points can host different uniform buffers name, so the cache must carry the index, and not the name.

I wondered too if we could set a binding point to always carry the same uniform block name, and in some cases where the UBO is heavily shared (like scene UBO or lights), this is definitely the way to go. But since a GPU usually carries less than 100 binding points, I prefer to leave the full liberty of binding for now. A PR for better handling of binding points is in my TODO list though :)